block/manager_test.go

```diff
@@ -549,6 +565,150 @@ func Test_publishBlock_ManagerNotProposer(t *testing.T) {
+	require.NoError(err)
+	require.NotNil(client)
+
+	vKey := ed25519.GenPrivKey()
+	validators := []*cmtypes.Validator{
+		{
+			Address:          vKey.PubKey().Address(),
+			PubKey:           vKey.PubKey(),
+			VotingPower:      int64(100),
+			ProposerPriority: int64(1),
+		},
+	}
+
+	lastState := types.State{}
+	lastState.ConsensusParams.Block = &cmproto.BlockParams{}
+	lastState.ConsensusParams.Block.MaxBytes = 100
+	lastState.ConsensusParams.Block.MaxGas = 100000
+	lastState.ConsensusParams.Abci = &cmproto.ABCIParams{VoteExtensionsEnableHeight: 0}
+	lastState.Validators = cmtypes.NewValidatorSet(validators)
+	lastState.NextValidators = cmtypes.NewValidatorSet(validators)
+	lastState.LastValidators = cmtypes.NewValidatorSet(validators)
+
+	mpool := mempool.NewCListMempool(cfg.DefaultMempoolConfig(), proxy.NewAppConnMempool(client, proxy.NopMetrics()), 0)
+	executor := state.NewBlockExecutor(vKey.PubKey().Address(), "test", mpool, proxy.NewAppConnConsensus(client, proxy.NopMetrics()), nil, 100, logger, state.NopMetrics())
+
+	signingKey, err := types.PrivKeyToSigningKey(vKey)
+	require.NoError(err)
+	m := &Manager{
+		lastState:    lastState,
+		lastStateMtx: new(sync.RWMutex),
+		blockCache:   NewBlockCache(),
+		executor:     executor,
+		store:        mockStore,
+		logger:       mockLogger,
+		genesis: &cmtypes.GenesisDoc{
+			ChainID:       "myChain",
+			InitialHeight: 1,
+			AppHash:       []byte("app hash"),
+		},
+		conf: config.BlockManagerConfig{
+			BlockTime:      time.Second,
+			LazyAggregator: false,
+		},
+		isProposer:  true,
+		proposerKey: signingKey,
+		metrics:     NopMetrics(),
+	}
+
+	t.Run("height should be updated after saving block responses", func(t *testing.T) {
+		mockStore.On("Height").Return(uint64(0))
+		signature := types.Signature([]byte{1, 1, 1})
+		block, err := executor.CreateBlock(0, &signature, abci.ExtendedCommitInfo{}, []byte{}, lastState)
+		require.NoError(err)
+		require.NotNil(block)
+		assert.Equal(uint64(0), block.Height())
+		dataHash, err := block.Data.Hash()
+		assert.NoError(err)
+		block.SignedHeader.DataHash = dataHash
+
+		// Update the signature on the block to current from last
+		voteBytes := block.SignedHeader.Header.MakeCometBFTVote()
+		signature, _ = vKey.Sign(voteBytes)
+		block.SignedHeader.Signature = signature
+		block.SignedHeader.Validators = lastState.Validators
+
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Millisecond)
+		defer cancel()
+
+		mockStore.On("GetBlock", mock.Anything, uint64(1)).Return(block, nil).Once()
+		mockStore.On("SaveBlock", mock.Anything, block, mock.Anything).Return(nil).Once()
+		mockStore.On("SaveBlockResponses", mock.Anything, uint64(0), mock.Anything).Return(nil).Once()
+		mockStore.On("UpdateState", mock.Anything, mock.Anything).Return(nil).Once()
+		mockStore.On("SetHeight", mock.Anything, uint64(0)).Return().Once()
+
+		err = m.publishBlock(ctx)
+		assert.ErrorContains(err, "unable to send header and block, context done")
```

Comment:
nitpick: Similarly to other comment, this is quite fragile - we should have this value as const somewhere or make a bit less-specific assertion.

Reply:
Documented in https://github.com/rollkit/rollkit/issues/1792